bin/envconfig

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# setup lsstsw environment
+#
+# source this file from your ~/.bashrc or ~/.zshrc
+#
+# relative to <lsstsw>/bin/
+
+print_error() {
+  >&2 echo -e "$@"
+}
+
+usage() {
+  if [[ ! -z "$1" ]]; then
+    print_error "$0: $1"
+  fi
+
+  # note that heredocs are prefixed with tab chars
+  echo "$(cat <<-EOF
+
+		Usage: $0  [-i] [-r] [-b] [-h]
+
+		Specific options:
+        -i          interactive, propose a list of available environment
+        -r REF      environment name
+        -b bXXXX    activate the environment already used for the build with id bXXXX
+        -h          show this message
+
+		EOF
+  )"
+
+  finish=true
+   
+}
+
+parse_args() {
+  local OPTIND
+  local opt
+
+  while getopts "hrib" opt; do
+    case "$opt" in
+    b)
+      shift;
+      BUILD_ID=$1
+      ;;
+    i)
+      INTERACTIVE=true
+      ;;
+    h)
+      usage
+      ;;
+    r)
+      shift;
+      ENVREF=$1
+      ;;
+    *)
+      usage "Unknown option: ${opt}"
+      ;;
+    esac
+  done
+  shift $((OPTIND-1))
+}
+
+##############
+
+INTERACTIVE=false
+ENVREF=""
+BUILD_ID=""
+
+parse_args "$@"
+
+if [[ !$finish ]]; then
+
+
+if [[ -z $ZSH_NAME ]]; then
```

Comment:
This repeats a lot of `setup.sh`.  Can the common code be factored out?

Reply:
I wanted something that introduces new functionalities and has no `setup` word in its name.
At the same time, I wanted to keep compatibility with the current way to work, so in my opinion, it is important to keep `setup.sh` as it.
If one day everybody is happy with `envconfig`, we can remove `setup.sh` from the tooling.